lib/Driver/Driver.cpp

```diff
@@ -2255,6 +2255,14 @@ bool Driver::handleImmediateArgs(const ArgList &Args, const ToolChain &TC) {
       commandLine.push_back(resourceDirArg->getValue());
     }
 
+    if (Args.hasFlag(options::OPT_static_executable,
+                     options::OPT_no_static_executable, false) ||
+        Args.hasFlag(options::OPT_static_stdlib,
+                     options::OPT_no_static_stdlib, false) ||
+        Args.hasArg(options::OPT_static)) {
+      commandLine.push_back("-static");
+    }
```

Comment:
I thought `-static-stdlib` was for statically linking the standard library but creating a dynamic executable/library.  Is that not the case?  If not, this makes sense, otherwise, this seems wrong because `-static` will will always do static linking.

Reply:
The linking is handled by the driver and `OPT_static` is not used in any form in the frontend currently. I'm happy to change it to something else, but the frontend does not seem to know `OPT_static_stdlib` or `OPT_static_executable`.

Reply:
Hmm, I suppose that my discomfort comes from the difference in what `-static` means when you are using `clang` and `swift`.  It just seems very easy to confuse someone (as opposed to many of the other arguments that have different spellings).

`-static`, `-static-executable` both imply creating a static binary.  `-static-stdlib` says that statically link the standard library but still still generate a dynamically linked binary.  As long `-static` is not forwarded from the frontend to the driver it will be okay I suppose, just confusing (non-Darwin targets use the linker driver (`clang`) to invoke the linker which will process `-static` as generating a static binary).  If you want to still go with this, I think that this deserves a comment explaining what is going on here and how this fits together.

Reply:
I added a new flag to the frontend `-use-static-resource-dir`